src/System.Management.Automation/utils/CryptoUtils.cs

```diff
@@ -867,14 +930,29 @@ internal PSRemotingCryptoHelperServer()
         internal override string EncryptSecureString(SecureString secureString)
         {
             // session!=null check required for DRTs TestEncryptSecureString* entries in CryptoUtilsTest/UTUtils.dll
-            // for newer clients, server will never initiate key exchange.
-            // for server, just the session key is required to encrypt/decrypt anything
-            if (Session is ServerRemoteSession session && session.Context.ClientCapability.ProtocolVersion >= RemotingConstants.ProtocolVersionWin8RTM)
+            bool initiateKeyExchange = true;
+
+            if (Session is ServerRemoteSession session)
             {
-                _rsaCryptoProvider.GenerateSessionKey();
+                Version clientProtocolVersion = session.Context.ClientCapability.ProtocolVersion;
+                if (clientProtocolVersion >= RemotingConstants.ProtocolVersion_2_4)
+                {
+                    // For client v2.4+, we no longer encrypt secure strings, but rely on the underlying secure transport to do the right thing.
```

Comment:
So does this mean that there is no check to see if a "secure" transport is used? What happens if someone is using a custom transport that isn't secure, uses something like WSMan without encryption (or weak NTLM encryption), etc?

Reply:
Thanks for taking a look!

The recommendation from the internal crypto board is to require secure transport for network connections. Perhaps it's possible to check if the WSMan connection is secure, but I doubt if that's doable in general. The plan is to update docs to add this requirement for network connections, but I don't know if this change would be considered a "regression" for users who are currently using PSRP over in-secure network transports.

The Security WG will discuss about this on Monday. I will post the update afterwards.

Reply:
Yea the fact that custom transports can do whatever they want makes it hard to guarantee a "secure transport". It's a pity that this won't be updated to a newer standard so that even on insecure transports SecureString's will stay encrypted but I understand not wanting to be in the same situation in x years time.

While I doubt many people are using custom transports it is a possibility and this change would mean SecureStrings are essentially plaintext.